Problem with gcc 4.8.5

Hello, I try to compile nginx-opentracing with gcc 4.8.5 on centos 7 and have this problem:
`./nginx-opentracing-0.7.0/opentracing/src/discover_span_context_keys.cpp: In constructor «ngx_opentracing::{anonymous}::HeaderKeyWriter::HeaderKeyWriter(ngx_pool_t*, std::vector<opentracing::v2::string_view>&)»:
./nginx-opentracing-0.7.0/opentracing/src/discover_span_context_keys.cpp:18:32: error: invalid initialization of non-const reference of type «std::vector<opentracing::v2::string_view>&» from an rvalue of type «<brace-enclosed initializer list>»
       : pool_{pool}, keys_{keys} {}`

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -21,6 +21,15 @@ jobs:
             ./ci/do_ci.sh system.testing
       - store_artifacts:
           path: /test-log
+  build_gcc48:
+    docker:
+      - image: ubuntu:18.04
+    steps:
+      - checkout
+      - run: ./ci/setup_build_environment.sh
+      - run: ./ci/install_gcc4.8.sh
+      - run: ./ci/install_opentracing.sh
+      - run: ./ci/do_ci.sh build
   docker_image:
     machine: true
     steps:
@@ -74,6 +83,7 @@ workflows:
             tags:
               only: /^v[0-9]+(\.[0-9]+)*$/
       - system_testing
+      - build_gcc48
       - docker_image
       - openresty_docker_image
       - module_binaries

--- a/ci/build_nginx_opentracing_module.sh
+++ b/ci/build_nginx_opentracing_module.sh
@@ -1,0 +1,15 @@
+#!/bin/bash
+
+set -e
+
+[ -z "${NGINX_VERSION}" ] && export NGINX_VERSION="1.13.12"
+
+cd "${BUILD_DIR}"
+wget -O nginx-release-${NGINX_VERSION}.tar.gz https://github.com/nginx/nginx/archive/release-${NGINX_VERSION}.tar.gz
+tar zxf nginx-release-${NGINX_VERSION}.tar.gz
+cd nginx-release-${NGINX_VERSION}
+ auto/configure \
+  --with-debug \
+  --add-dynamic-module="${SRC_DIR}/opentracing"
+make VERBOSE=1
+

--- a/ci/build_static_opentracing.sh
+++ b/ci/build_static_opentracing.sh
@@ -1,5 +1,7 @@
 #!/bin/bash
 
+set -e
+
 pushd "${BUILD_DIR}"
 git clone -b v$OPENTRACING_VERSION https://github.com/opentracing/opentracing-cpp.git
 cd opentracing-cpp

--- a/ci/do_ci.sh
+++ b/ci/do_ci.sh
@@ -5,7 +5,7 @@ set -e
 [ -z "${SRC_DIR}" ] && export SRC_DIR="`pwd`"
 [ -z "${BUILD_DIR}" ] && export BUILD_DIR=/build
 [ -z "${MODULE_DIR}" ] && export MODULE_DIR=/modules
-
+mkdir -p "${BUILD_DIR}"
 
 if [[ "$1" == "system.testing" ]]; then
   docker build -t nginx-opentracing-test/nginx -f Dockerfile-test .
@@ -15,6 +15,9 @@ if [[ "$1" == "system.testing" ]]; then
   docker build -t nginx-opentracing-test/grpc-backend .
   cd -
   PYTHONPATH=environment/grpc python3 nginx_opentracing_test.py
+  exit 0
+elif [[ "$1" == "build" ]]; then
+  ./ci/build_nginx_opentracing_module.sh
   exit 0
 elif [[ "$1" == "module.binaries" ]]; then
   mkdir -p "${BUILD_DIR}"

--- a/ci/do_ci.sh
+++ b/ci/do_ci.sh
@@ -5,7 +5,6 @@ set -e
 [ -z "${SRC_DIR}" ] && export SRC_DIR="`pwd`"
 [ -z "${BUILD_DIR}" ] && export BUILD_DIR=/build
 [ -z "${MODULE_DIR}" ] && export MODULE_DIR=/modules
-mkdir -p "${BUILD_DIR}"
 
 if [[ "$1" == "system.testing" ]]; then
   docker build -t nginx-opentracing-test/nginx -f Dockerfile-test .
@@ -17,6 +16,7 @@ if [[ "$1" == "system.testing" ]]; then
   PYTHONPATH=environment/grpc python3 nginx_opentracing_test.py
   exit 0
 elif [[ "$1" == "build" ]]; then
+  mkdir -p "${BUILD_DIR}"
   ./ci/build_nginx_opentracing_module.sh
   exit 0
 elif [[ "$1" == "module.binaries" ]]; then

--- a/ci/install_gcc4.8.sh
+++ b/ci/install_gcc4.8.sh
@@ -1,0 +1,13 @@
+#!/bin/bash
+
+set -e
+apt-get update 
+apt-get install --no-install-recommends --no-install-suggests -y \
+          software-properties-common
+add-apt-repository -y ppa:ubuntu-toolchain-r/test
+apt-get update
+apt-get install --no-install-recommends --no-install-suggests -y \
+          gcc-4.8 \
+          g++-4.8
+update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-4.8 5
+update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-4.8 5

--- a/ci/install_opentracing.sh
+++ b/ci/install_opentracing.sh
@@ -1,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+[ -z "${OPENTRACING_VERSION}" ] && export OPENTRACING_VERSION="v1.5.0"
+
+# Build OpenTracing
+cd /
+git clone -b ${OPENTRACING_VERSION} https://github.com/opentracing/opentracing-cpp.git
+cd opentracing-cpp
+mkdir .build && cd .build
+cmake \
+  -DBUILD_MOCKTRACER=OFF \
+  -DBUILD_TESTING=OFF \
+  ..
+make && make install

--- a/ci/setup_build_environment.sh
+++ b/ci/setup_build_environment.sh
@@ -15,4 +15,4 @@ apt-get install --no-install-recommends --no-install-suggests -y \
                 ssh \
                 wget \
                 libpcre3 libpcre3-dev \
-                zlib1g-dev \
+                zlib1g-dev

--- a/opentracing/src/discover_span_context_keys.cpp
+++ b/opentracing/src/discover_span_context_keys.cpp
@@ -15,7 +15,7 @@ namespace {
 class HeaderKeyWriter : public opentracing::HTTPHeadersWriter {
  public:
   HeaderKeyWriter(ngx_pool_t* pool, std::vector<opentracing::string_view>& keys)
-      : pool_{pool}, keys_{keys} {}
+      : pool_{pool}, keys_(keys) {}
 
   opentracing::expected<void> Set(
       opentracing::string_view key,

--- a/opentracing/src/opentracing_directive.cpp
+++ b/opentracing/src/opentracing_directive.cpp
@@ -138,7 +138,7 @@ char *propagate_opentracing_context(ngx_conf_t *cf, ngx_command_t * /*command*/,
 
   auto old_args = cf->args;
 
-  ngx_str_t args[] = {ngx_string("proxy_set_header"), {}, {}};
+  ngx_str_t args[] = {ngx_string("proxy_set_header"), ngx_str_t(), ngx_str_t()};
   ngx_array_t args_array;
   args_array.elts = static_cast<void *>(&args);
   args_array.nelts = 3;
@@ -185,7 +185,7 @@ char *propagate_fastcgi_opentracing_context(ngx_conf_t *cf,
 
   auto old_args = cf->args;
 
-  ngx_str_t args[] = {ngx_string("fastcgi_param"), {}, {}};
+  ngx_str_t args[] = {ngx_string("fastcgi_param"), ngx_str_t(), ngx_str_t()};
   ngx_array_t args_array;
   args_array.elts = static_cast<void *>(&args);
   args_array.nelts = 3;
@@ -229,7 +229,7 @@ char *propagate_grpc_opentracing_context(ngx_conf_t *cf, ngx_command_t *command,
 
   auto old_args = cf->args;
 
-  ngx_str_t args[] = {ngx_string("grpc_set_header"), {}, {}};
+  ngx_str_t args[] = {ngx_string("grpc_set_header"), ngx_str_t(), ngx_str_t()};
   ngx_array_t args_array;
   args_array.elts = static_cast<void *>(&args);
   args_array.nelts = 3;

--- a/opentracing/src/span_context_querier.cpp
+++ b/opentracing/src/span_context_querier.cpp
@@ -32,7 +32,7 @@ ngx_str_t SpanContextQuerier::lookup_value(ngx_http_request_t* request,
                 "no opentracing context value found for span context key %V "
                 "for request %p",
                 &ngx_key, request);
-  return {};
+  return ngx_str_t();
 }
 
 //------------------------------------------------------------------------------
@@ -43,7 +43,7 @@ class SpanContextValueExpander : public opentracing::HTTPHeadersWriter {
  public:
   explicit SpanContextValueExpander(
       std::vector<std::pair<std::string, std::string>>& span_context_expansion)
-      : span_context_expansion_{span_context_expansion} {}
+      : span_context_expansion_(span_context_expansion) {}
 
   opentracing::expected<void> Set(
       opentracing::string_view key,

--- a/opentracing/src/span_context_querier.h
+++ b/opentracing/src/span_context_querier.h
@@ -17,7 +17,7 @@ extern "C" {
 namespace ngx_opentracing {
 class SpanContextQuerier {
  public:
-  SpanContextQuerier() noexcept = default;
+  SpanContextQuerier() noexcept {}
 
   ngx_str_t lookup_value(ngx_http_request_t* request,
                          const opentracing::Span& span,

